pkgs/development/python-modules/validphys2/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib
+, buildPythonPackage
+, nnpdf
+, reportengine
+}:
+
+buildPythonPackage rec {
+  pname = "validphys2";
+  version = "4.0";
+  format = "setuptools";
+
+  inherit (nnpdf) src;
+
+  prePatch = ''
+    cd validphys2
```

Comment:
```suggestion
  sourceRoot = "${src.name}/validphys2";
```

Reply:
`sourceRoot` is broken by design. Your varian will break when overriding with a local path like `src = lib.cleanSources ./nnpdf` and possibly also for overriding to `fetchurl` and `fetchgit`.

Reply:
You should at least use `postUnpack`.

Reply:
`postUnpack` is called before `cd $sourceRoot`
https://github.com/NixOS/nixpkgs/blob/ea69f79829972187ecf6c1eaede4ce7a15d81736/pkgs/stdenv/generic/setup.sh#L1364-L1374